test/integration_tests/python/test_uploads.py

```diff
@@ -134,8 +149,22 @@ def test_uid_upload(data_builder, file_form, as_admin, as_user, as_public):
     r = as_user.post('/upload/uid', files=file_form(*uid_files, meta=uid_meta))
     assert r.status_code == 403
 
+    # TODO figure out why api.dao.hierarchy._group_id_fuzzy_match is NOT called below
```

Comment:
Is that function not called at all, or just not the fuzzy match part? I can look into this too. I don't think it holds up this merging, though.